eclipse-che/che-theia
# Align che-theia with upstream changes

### What does this PR do?
- Moves che-theia to ES2017.
- Provides adaptation to upstream changes.
- Pin `che-theia` to `theia` commit before new breaking change (https://github.com/eclipse-theia/theia/commit/23e09c2df53e6b8c57ea14c1baf2ffbaa6555408). Adaption for those changes will be provided within a separate PR.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20366
https://github.com/eclipse/che/issues/20360

### How to test this PR?

- the PR fixes deploy of remote plugins, please see `Steps to reproduce` [here](https://github.com/eclipse/che/issues/20360)
- activation of VS Code extensions should be fixed, please see the comment [here](https://github.com/eclipse/che/issues/20366#issuecomment-909253757).
- all other functionality should work as before breaking changes in upstream.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)



### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable
